src/cryptography/x509/extensions.py

```diff
@@ -156,8 +156,17 @@ def __eq__(self, other):
         if not isinstance(other, AuthorityKeyIdentifier):
             return NotImplemented
 
+        if self.key_identifier is None:
```

Comment:
The main problem here is that `constant_time.bytes_eq()` raises exceptions if either argument is None. If you have any ideas on how to avoid the branches without introducing more complexity, I'd love to know.